File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.kt
class Solution {
    fun solution(s: String, skip: String, index: Int): String {
        val answer = StringBuilder()
        
        for(c in s) {
            answer.append(charChanger(c, skip, index))
        }
        
        return answer.toString()
    }

    fun charChanger(c: Char, skip: String, index: Int): Char {
        val skipToArr = skip.toCharArray()
        var cToAscii = c.code
        var count = 0

        while(true) {
            if(cToAscii+1 > 122) cToAscii = 97 else cToAscii++
            
            if(!(skipToArr.contains((cToAscii).toChar()))) count++
            
            if (count == index) break
        }
        return cToAscii.toChar()
    }
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.swift
func solution(_ s:String, _ skip:String, _ index:Int) -> String {
    var answer = ""
    var skipToAsciis = stringToAsciis(skip)
    
    for i in stringToAsciis(s) {
        var ascii = i
        var count = 0
        
        while count < index {
            ascii += (ascii < 122) ? 1 : -25
            count += (skipToAsciis.contains(ascii)) ? 0 : 1
        }
        
        answer += asciiToString(ascii)
    }
    
    return answer
}

func stringToAsciis(_ s: String) -> [Int] {
    return s.map { Int(UnicodeScalar(String($0))!.value) }
}

func asciiToString(_ i: Int) -> String {
    return String(UnicodeScalar(i)!)
}